mithril-aggregator/src/http_server/routes/epoch_routes.rs

```diff
@@ -179,6 +180,95 @@ mod tests {
+            ..AggregatorEpochSettings::dummy()
+        };
+        let next_epoch_settings = AggregatorEpochSettings {
+            protocol_parameters: ProtocolParameters::new(102, 20, 0.5),
+            ..AggregatorEpochSettings::dummy()
+        };
+        let upcoming_epoch_settings = AggregatorEpochSettings {
+            protocol_parameters: ProtocolParameters::new(103, 30, 0.5),
+            ..AggregatorEpochSettings::dummy()
+        };
+
+        let epoch_service = FakeEpochService::with_data(
+            Epoch(1),
+            &current_epoch_settings,
+            &next_epoch_settings,
+            &upcoming_epoch_settings,
+            &fake_data::signers_with_stakes(5),
+            &fake_data::signers_with_stakes(3),
+        );
+
+        let message = get_epoch_settings_message(
+            Arc::new(RwLock::new(epoch_service)),
+            SignedEntityConfig::dummy(),
+        )
+        .await
+        .unwrap();
+
+        // TODO: What we expect  ? see #1963 Discrepancy of protocol parameters in epoch settings and pending certificates routes
+        // https://github.com/input-output-hk/mithril/issues/1963
+        assert_eq!(
+            message.protocol_parameters,
+            next_epoch_settings.protocol_parameters
+        );
+        assert_eq!(
+            message.next_protocol_parameters,
+            upcoming_epoch_settings.protocol_parameters
+        );
+    }
+
+    #[tokio::test]
+    async fn get_epoch_settings_message_retrieves_signing_configuration_from_epoch_service() {
+        let current_epoch_settings = AggregatorEpochSettings {
+            cardano_transactions_signing_config: CardanoTransactionsSigningConfig::new(
+                BlockNumber(100),
+                BlockNumber(15),
+            ),
+            ..AggregatorEpochSettings::dummy()
+        };
+        let next_epoch_settings = AggregatorEpochSettings {
+            cardano_transactions_signing_config: CardanoTransactionsSigningConfig::new(
+                BlockNumber(200),
+                BlockNumber(15),
+            ),
+            ..AggregatorEpochSettings::dummy()
+        };
+
+        let epoch_service = FakeEpochService::with_data(
+            Epoch(1),
+            &current_epoch_settings,
+            &next_epoch_settings,
+            &AggregatorEpochSettings::dummy(),
+            &fake_data::signers_with_stakes(5),
+            &fake_data::signers_with_stakes(3),
+        );
+
+        let message = get_epoch_settings_message(
+            Arc::new(RwLock::new(epoch_service)),
+            SignedEntityConfig::dummy(),
+        )
+        .await
+        .unwrap();
+
+        // TODO: What we expect  ? see #1963 Discrepancy of protocol parameters in epoch settings and pending certificates routes
+        // https://github.com/input-output-hk/mithril/issues/1963
+        // TODO if they need to be aligned with protocol parameters, we may assert in the same test
```

Comment:
Same as above, this can be fixed now that #1963 is closed

Reply:
Done ✅